packages/gatsby/src/schema/schema.js

```diff
@@ -591,6 +649,65 @@ const addTypeToRootQuery = ({ schemaComposer, typeComposer }) => {
   })
 }
 
+const parseTypes = ({
+  doc,
+  plugin,
+  createdFrom,
+  schemaComposer,
+  parentSpan,
+}) => {
+  const types = []
+  doc.definitions.forEach(def => {
+    const name = def.name.value
+    checkIsAllowedTypeName(name)
+
+    if (schemaComposer.has(name)) {
+      // We don't check if ast.kind matches composer type, but rely
+      // that this will throw when something is wrong and get
+      // reported by `reportParsingError`.
+
+      // Keep the original type composer around
+      const typeComposer = schemaComposer.get(name)
+      // Only allow user to extend an already existing type
+      if (!plugin || plugin.name === `default-site-plugin`) {
```

Comment:
Am I missing something or is this only checked for SDL types?  What about plugin defined type builder or graphql-js types?

Reply:
You're right, this was wrong.